### ISSUE TITLE
Problem with Restya via Docker and Plesk

### ISSUE THREAD

**Issue by KarlFrederik** [RestyaPlatform/board#4018]
I have a Problem with Restyaboard on Docker and Plesk.
I've installed Restyaboard and PostgreSQL via Docker on my Server with Plesk and now I have one major problem. 
The console of Restyaboard says:

> sed: -e expression #1, char 73: unknown option to `s'

the console of Postgres says:

> 2020-03-31 16:41:17.991 UTC [107] LOG:  invalid length of startup packet

What can I do?
Thanks for Helping

My Hardware and Software:
Intel(R) Xeon(R) CPU E5-2630
Plesk Obsidian v18.0.25_build1800200325.19 os_Ubuntu 18.04
Ubuntu 18.04.4 LTS
Hosted at Strato Germany

**Comment by sridhar391** [comment]
@KarlFrederik Sorry for your inconvenience, can you please tell us after executing
which command the below error occurred

sed: -e expression #1, char 73: unknown option to `s'

**Comment by KarlFrederik** [comment]
> @KarlFrederik Sorry for your inconvenience, can you please tell us after executing
> which command the below error occurred
> 
> sed: -e expression #1, char 73: unknown option to `s'

I executed no command. When I start the Container in Docker, it occurred.
Here is a Screenshot of that:
![Bildschirmfoto vom 2020-04-02 14-16-15](https://user-images.githubusercontent.com/42519173/78248274-9b032580-74ec-11ea-82ea-c010b8d59504.png)

**Comment by rrjanbiah** [comment]
`sed` usage with `e` is tracked to `docker-scripts/docker-entrypoint.sh` 

@BlackDark has recently fixed some docker related issues. So, not sure if it is anything related to that. 

@BlackDark Do you have any comments here, please?

**Comment by BlackDark** [comment]
@KarlFrederik do your postgres variables contain `/`?
This probably breaks the sed command in the startup script

**Comment by KarlFrederik** [comment]
> @KarlFrederik do your postgres variables contain `/`?
> This probably breaks the sed command in the startup script

I don't know. I don't have much knowledge about that.
I send you a screenshot of my Variables.

![Bildschirmfoto vom 2020-04-03 19-05-35](https://user-images.githubusercontent.com/42519173/78386733-5eaff200-75de-11ea-9e02-ca7a20db63f4.png)

**Comment by KarlFrederik** [comment]
> No i mean the variables you have to specify for restyaboard: https://github.com/RestyaPlatform/board/blob/dev/docker-compose.prod.yml#L6
> 
> ```
>       - POSTGRES_HOST=postgres
>       - POSTGRES_USER=admin
>       - POSTGRES_PASSWORD=admin
>       - POSTGRES_DB=restyaboard
> ```
> 
> how do they look like in your deployment? or didn't you set them?

I think I never set them. Where can I change or set them?
Maybe that's the problem.

**Comment by BlackDark** [comment]
https://docs.plesk.com/de-DE/obsidian/administrator-guide/75823/#container-settings

**Comment by KarlFrederik** [comment]
> https://docs.plesk.com/de-DE/obsidian/administrator-guide/75823/#container-settings

My Variables at Restyaboard are these:
```

- POSTGRES_HOST=h2872773.stratoserver.net:5433/
- POSTGRES_PORT=5432
- POSTGRES_USER=postgres
- POSTGRES_PASSWORD=password1
- POSTGRES_DB=restyaboard
```

I set the password in the postgres container to password1 so I can remember it better.
It could be the problem that I set up postgres wrong.
Could someone send me a link to a good tutorial?
That would be nice.

**Comment by BlackDark** [comment]
Ah this could be probably caused by the `POSTGRES_HOST`.
Could you remove the trailing `/` and check if its working then?

**Comment by KarlFrederik** [comment]
That changes the error.
Now the console says:
`psql: could not translate host name "h2872773.stratoserver.net:5433" to address: Name or service not known`

**Comment by BlackDark** [comment]
Not sure how deployment is done in plesk.

You could try following:

- Change the host and port like this

```
- POSTGRES_HOST=h2872773.stratoserver.net
- POSTGRES_PORT=5433
- POSTGRES_USER=postgres
- POSTGRES_PASSWORD=password1
- POSTGRES_DB=restyaboard
``` 

- or if it set up like a docker-compose and therefore internal docker referencing works you could try following (use containername as host and internal port). Maybe you could check through the docs if in plesk the deployed container share a network or similiar.

```
- POSTGRES_HOST=postgres
- POSTGRES_PORT=5432
- POSTGRES_USER=postgres
- POSTGRES_PASSWORD=password1
- POSTGRES_DB=restyaboard
```

**Comment by KarlFrederik** [comment]
Now there is another problem.
```
psql: could not connect to server: Connection refused
	Is the server running on host "h2872773.stratoserver.net" (81.169.173.113) and accepting
	TCP/IP connections on port 5432?
```
Is it a Port forwading error?

**Comment by BlackDark** [comment]
Did you also update the POSTGRES_PORT to 5433?
Because the error says something about port 5432

**Comment by KarlFrederik** [comment]
Oh, 
yep. Now it can connect.
Thats the new error
`psql: FATAL:  database "restyaboard" does not exist`
The problem is, that i created the database and i can connect with it via the Termina:
![Bildschirmfoto vom 2020-04-07 18-22-28](https://user-images.githubusercontent.com/42519173/78694323-c88d0c00-78fc-11ea-9c3b-028c38bd8772.png)
I can connect, but restyaboard doesn'T find the database.

**Comment by BlackDark** [comment]
You created the database manually?
This should be done automically by restyaboard.

Currently i am updating the db init for docker but still this is a step which is done initially by the container. Otherwise you will have to add all tables etc yourself with the SQL file here: https://github.com/RestyaPlatform/board/tree/dev/sql (empty data file).

In general what restyaboard does initially (copied mostly to docker container) you can find here: https://github.com/RestyaPlatform/board/blob/dev/restyaboard.sh#L1014 (yep not the best and overloaded. @rrjanbiah do you plan to split the startup script into multiple parts to make it more modular? The script with nearly 2000 lines is painful.)

So to sum up:

- Restyaboard handles the database creation itself. Including creating the user + database + extension + relations.
- If you want to create DB yourself you have to make sure to create the User for restyaboard database and apply the empty db schema mentioned above

**Comment by rrjanbiah** [comment]
> @rrjanbiah do you plan to split the startup script into multiple parts to make it more modular? The script with nearly 2000 lines is painful.)

@BlackDark Yes sure.

**Comment by KarlFrederik** [comment]
> You created the database manually?
> This should be done automically by restyaboard.
> 
> Currently i am updating the db init for docker but still this is a step which is done initially by the container. Otherwise you will have to add all tables etc yourself with the SQL file here: https://github.com/RestyaPlatform/board/tree/dev/sql (empty data file).
> 
> In general what restyaboard does initially (copied mostly to docker container) you can find here: https://github.com/RestyaPlatform/board/blob/dev/restyaboard.sh#L1014 (yep not the best and overloaded. @rrjanbiah do you plan to split the startup script into multiple parts to make it more modular? The script with nearly 2000 lines is painful.)
> 
> So to sum up:
> 
>     * Restyaboard handles the database creation itself. Including creating the user + database + extension + relations.
> 
>     * If you want to create DB yourself you have to make sure to create the User for restyaboard database and apply the empty db schema mentioned above

I didn't plan to create the database myself. But it didn't created it atomaticly. So I thought when i create the database myself, maybe it works.

**Comment by KarlFrederik** [comment]
I reinstalled it and now there is this error:
`==> /var/log/nginx/error.log <==
Starting periodic command scheduler: cron.
Starting nginx: nginx.
Starting Postfix Mail Transport Agent: postfix.
==> /var/log/nginx/access.log <==

==> /var/log/nginx/error.log <==`
![Bildschirmfoto vom 2020-04-08 17-27-54](https://user-images.githubusercontent.com/42519173/78802718-6d235280-79be-11ea-8eaa-791183a5ccb3.png)

**Comment by BlackDark** [comment]
> I reinstalled it and now there is this error:
> `==> /var/log/nginx/error.log <==
> Starting periodic command scheduler: cron.
> Starting nginx: nginx.
> Starting Postfix Mail Transport Agent: postfix.
> ==> /var/log/nginx/access.log <==
> 
> ==> /var/log/nginx/error.log <==`
> ![Bildschirmfoto vom 2020-04-08 17-27-54](https://user-images.githubusercontent.com/42519173/78802718-6d235280-79be-11ea-8eaa-791183a5ccb3.png)

Do you use the latest dev version from docker hub? Maybe you have to make sure to pull the newest image via plesk (maybe theres a toggle?)

the `/var/log/nginx/error.log` is not an error, it is just an information of the used log file.

```bash
❯ docker-compose up -d  
Creating network "restyaboard-fork_default" with the default driver
Creating restyaboard-fork_fakesmtp2_1   ... done
Creating restyaboard-fork_postgres_1   ... done
Creating restyaboard-fork_smtp_relay_1  ... done
Creating restyaboard-fork_restyaboard_1 ... done
    ~/projects/github/restyaboard-fork    optimize-docker-db                                                           
❯ docker-compose logs  restyaboard  
Attaching to restyaboard-fork_restyaboard_1
restyaboard_1  | Starting restya ...
restyaboard_1  | Trying to connect to postgres ...
restyaboard_1  | postgres:5432 - no response
restyaboard_1  | Trying to connect to postgres ...
restyaboard_1  | postgres:5432 - accepting connections
restyaboard_1  | Connection established!
restyaboard_1  | Creating PostgreSQL user and database...
restyaboard_1  | NOTICE:  role "restya" does not exist, skipping
restyaboard_1  | CREATE ROLE
restyaboard_1  | CREATE DATABASE
restyaboard_1  | CREATE EXTENSION
restyaboard_1  | NOTICE:  extension "plpgsql" already exists, skipping
restyaboard_1  | COMMENT
restyaboard_1  | Importing empty SQL... (pipe output to /tmp/restya_initdb.log)
restyaboard_1  | psql:/var/lib/nginx/html/sql/restyaboard_with_empty_data.sql:29: ERROR:  must be owner of extension plpgsql
restyaboard_1  | psql:/var/lib/nginx/html/sql/restyaboard_with_empty_data.sql:7382: WARNING:  no privileges could be revoked for "public"
restyaboard_1  | psql:/var/lib/nginx/html/sql/restyaboard_with_empty_data.sql:7383: WARNING:  no privileges could be revoked for "public"
restyaboard_1  | psql:/var/lib/nginx/html/sql/restyaboard_with_empty_data.sql:7384: WARNING:  no privileges were granted for "public"
restyaboard_1  | psql:/var/lib/nginx/html/sql/restyaboard_with_empty_data.sql:7385: WARNING:  no privileges were granted for "public"
restyaboard_1  | Succesfully imported initial data.
restyaboard_1  | ==> /var/log/nginx/access.log <==
restyaboard_1  | 
restyaboard_1  | ==> /var/log/nginx/error.log <==
restyaboard_1  | 
restyaboard_1  | ==> /var/log/cron.log <==
restyaboard_1  | crond: crond (busybox 1.31.1) started, log level 8
restyaboard_1  | 
restyaboard_1  | ==> /var/log/php7/error.log <==
restyaboard_1  | [09-Apr-2020 09:36:53] NOTICE: fpm is running, pid 27
restyaboard_1  | [09-Apr-2020 09:36:53] NOTICE: ready to handle connections
```

**Comment by BlackDark** [comment]
@KarlFrederik still problems or can this issue be closed?

**Comment by KarlFrederik** [comment]
I still have problems.
When I start the software, I get this message in the console:
`
Starting periodic command scheduler: cron.
Starting nginx: nginx.
Starting Postfix Mail Transport Agent: postfix.
==> /var/log/nginx/access.log <==

==> /var/log/nginx/error.log <==
Starting periodic command scheduler: cron.
Starting nginx: nginx.
Starting Postfix Mail Transport Agent: postfix.
==> /var/log/nginx/access.log <==

==> /var/log/nginx/error.log <==
`

i waited a few days and nothing changed.

**Comment by BlackDark** [comment]
Could you please try to use the `dev` tag version of the docker image for restyaboard?

**Comment by KarlFrederik** [comment]
I installed it now and there aren't much variables pre-installed.
What is RESTYA_DB_USERNAME ?

**Comment by BlackDark** [comment]
Check here: https://github.com/RestyaPlatform/board#docker-configuration

You can use for testing `restya/restya`

**Comment by KarlFrederik** [comment]
Now this error Ocured:

> Starting restya ...
Trying to connect to postgres ...
h2872773.stratoserver.net:5433 - accepting connections
Connection established!
Creating PostgreSQL user and database...
PostgreSQL user creation failed with error code 1 (PostgreSQL user creation failed with error code 35)
Failed to initialize DB for restyaboard.
ERROR:  current user cannot be dropped

**Comment by BlackDark** [comment]
Did you set `POSTGRES_ADMIN_USER ` and `POSTGRES_ADMIN_PASS ` to the admin user you created in postgres?

Probably `postgres/password1`

**Comment by KarlFrederik** [comment]
> Did you set `POSTGRES_ADMIN_USER ` and `POSTGRES_ADMIN_PASS ` to the admin user you created in postgres?
> 
> Probably `postgres/password1`

Now this is a problem:


> Starting restya ...
> 
> Trying to connect to postgres ...
> h2872773.stratoserver.net:5433 - accepting connections
> Connection established!
> Creating PostgreSQL user and database...
> CREATE ROLE
> CREATE DATABASE
> CREATE EXTENSION
> COMMENT
> Importing empty SQL... (pipe output to /tmp/restya_initdb.log)
> Succesfully imported initial data.
> ==> /var/log/nginx/access.log <==
> 
> ==> /var/log/nginx/error.log <==
> 
> ==> /var/log/cron.log <==
> crond: crond (busybox 1.31.1) started, log level 8
> 
> ==> /var/log/php7/error.log <==
> [17-Apr-2020 20:35:14] NOTICE: fpm is running, pid 25
> [17-Apr-2020 20:35:14] NOTICE: ready to handle connections
> NOTICE:  role "restya" does not exist, skipping
> NOTICE:  extension "plpgsql" already exists, skipping
> psql:/var/lib/nginx/html/sql/restyaboard_with_empty_data.sql:29: ERROR:  must be owner of extension plpgsql
> psql:/var/lib/nginx/html/sql/restyaboard_with_empty_data.sql:7382: WARNING:  no privileges could be revoked for "public"
> psql:/var/lib/nginx/html/sql/restyaboard_with_empty_data.sql:7383: WARNING:  no privileges could be revoked for "public"
> psql:/var/lib/nginx/html/sql/restyaboard_with_empty_data.sql:7384: WARNING:  no privileges were granted for "public"
> psql:/var/lib/nginx/html/sql/restyaboard_with_empty_data.sql:7385: WARNING:  no privileges were granted for "public"
>

**Comment by BlackDark** [comment]
That's probalby no error check if it is working. Open your browser to your exposed port / website. Should be working.

**Comment by KarlFrederik** [comment]
Which could be my exposed port?

**Comment by BlackDark** [comment]
https://support.plesk.com/hc/en-us/articles/115004844374-How-to-proxy-requests-from-domains-to-Docker-container-in-Plesk-

Restyaboard exposes its port on 80
